docs/proposals/004-orchestration-refresh-certs.md

```diff
@@ -0,0 +1,266 @@
+`v1beta2.k8sd.io/refresh-certificates` annotation is applied.
+
+The status of the certificate refresh process will be shared via the `Cluster`
+object by emitting events. The events that the controller can emit are:
+- `RefreshCertsInProgress`: The certificate refresh process has started.
+- `RefreshCertsDone`: The certificate refresh process has finished successfully.
+- `RefreshCertsFailed`: The certificate refresh process has failed.
+
+The controller should perform the following steps:
+1. Retrieve the `CK8sControlPlane` object owned by the `Cluster` object.
+2. Emit the `RefreshCertsInProgress` event for the `Cluster` object.
+3. Trigger a certificate refresh for the control plane nodes by annotating the
+   `CK8sControlPlane` object with the `v1beta2.k8sd.io/refresh-certificates`
+   annotation.
+4. Wait for the certificates to be refreshed on the control plane nodes. The
+   controller should check the `v1beta2.k8sd.io/refresh-certificates-status`
+   annotation to determine when the certificates have been refreshed.
+5. If the refresh is successful, the controller proceeds to the
+   `MachineDeployment` objects.
+6. For each `MachineDeployment` object, trigger a certificate refresh for the
+   worker nodes by annotating the `MachineDeployment` object with the
+   `v1beta2.k8sd.io/refresh-certificates` annotation.
+7. Wait for the certificates to be refreshed on the worker nodes, checking the
+   `v1beta2.k8sd.io/refresh-certificates-status` annotation.
+8. If the refresh is successful, the controller emits the `RefreshCertsDone`
+   event for the `Cluster` object.
+
+### MachineDeployment Controller
+
+We also need to add a new controller, `MachineDeployCertificatesReconciler`, to
+the bootstrap provider. This controller will watch for `MachineDeployment`
+objects and trigger a certificate refresh for all the worker nodes in the
+cluster when the `v1beta2.k8sd.io/refresh-certificates` annotation is present.
+
+The controller should perform the following steps:
+1. List all machines owned by the `MachineDeployment` object and filter out the
+   control plane machines.
+2. Emit the `RefreshCertsInProgress` event for the `MachineDeployment` object.
+3. For each machine, trigger the certificate refresh by annotating the machine
+   with the `v1beta2.k8sd.io/refresh-certificates` annotation.
+4. Wait for the certificates to be refreshed on that machine. The controller
+   should check the `v1beta2.k8sd.io/refresh-certificates-status` annotation
+   to know when the certificates have been refreshed.
+5. If the refresh is successful, the controller moves to the next machine.
+
+The status of the certificate refresh process will be shared via the
+`MachineDeployment` object in the same way as the `Cluster` controller.
+
+## ControlPlane Provider Changes
+<!--
+This section MUST mention any changes to the controlplane provider.
+-->
+
+A controller `CK8sCertificatesReconciler` will be added to the control plane
```

Comment:
```suggestion
A controller `ControlPlaneCertificatesReconciler` will be added to the control plane
```
Is this a copy paste error or intentional?

Reply:
That was intentional, but I like your naming better. Thanks!